src/NativeScript/ObjC/ObjCTypes.mm

```diff
@@ -202,35 +147,13 @@ JSValue toValue(ExecState* execState, id object, Structure* (^structureResolver)
         return jsNull();
     }
 
-#if defined(__LP64__) && __LP64__
-    // There is a bug in the Objective-C runtime on 64 bit architectures, which seems to be fixed in iOS 8.
-    // We cannot use the association API with tagged pointers, so we fallback to an external map.
-    // This workaround is kept for iOS 8 64bit, because it is faster there too and the code is easier to test/maintain.
-    // For more information on tagged pointers see "objc-internal.h" in opensource.apple.com
-
-    // This check is the same as _objc_isTaggedPointer, which is a private function.
-    if (reinterpret_cast<intptr_t>(object) < 0) {
-        NativeScript::GlobalObject* globalObject = jsCast<NativeScript::GlobalObject*>(execState->lexicalGlobalObject());
-        if (ObjCWrapperObject* wrapper = globalObject->taggedPointers().get(object)) {
-            return wrapper;
-        }
-
-        ObjCWrapperObject* wrapper = ObjCWrapperObject::create(execState->vm(), structureResolver(), object, jsCast<GlobalObject*>(execState->lexicalGlobalObject()));
-        globalObject->taggedPointers().set(object, wrapper);
-        return wrapper;
-    } else {
-#endif
-        TNSValueWrapper* valueWrapper = static_cast<TNSValueWrapper*>(objc_getAssociatedObject(object, execState->lexicalGlobalObject()->JSCell::vm()));
-        if (JSObject* wrapper = valueWrapper.value) {
-            return wrapper;
-        }
-
-        ObjCWrapperObject* wrapper = ObjCWrapperObject::create(execState->vm(), structureResolver(), object, jsCast<GlobalObject*>(execState->lexicalGlobalObject()));
-        [TNSValueWrapper attachValue:wrapper
-                              toHost:object];
+    auto globalObject = jsCast<GlobalObject*>(execState->lexicalGlobalObject());
+    if (JSObject* wrapper = globalObject->interop()->objectMap().get(object)) {
         return wrapper;
```

Comment:
Consider enforcing the invariant that the `wrappedObject` is the same as `object` if possible.